python/ray/ml/config.py

```diff
@@ -0,0 +1,9 @@
+from typing import Any, Dict
+
+
+# Right now, RunConfig is just an arbitrary dict that specifies tune.run
+# kwargs.
+# TODO: After Tuner is implemented, make this into an actual dataclass
+RunConfig = Dict[str, Any]
+
+ScalingConfig = Dict[str, Any]
```

Comment:
Same, let's document the schema.

Reply:
Following up from our previous discussion, a schema specifically for DataParallel scaling makes sense, but if we were to introduce a ModelParallelTrainer for example, this will likely be a very different scaling config. So I'm not sure if a generic schema for all types of Trainers makes sense.

Specifically for Data Parallel scaling config, this is documented in the follow up PR.

Also cc @richardliaw

Reply:
Yes, the contents of the scaling config dict are trainer-specific. My proposal here is to document here that for the contents, users have to look into the respective trainer classes.

To make this more general, we should move the actual ScalingConfig dataclass into the Trainer class:

```
class Trainer(abc.ABC):

    @dataclass
    class ScalingConfig:
        """Documentation for trainer-specific scaling config"""
        pass
```

specific trainers can then override this class:

```
class DataParallelTrainer(Trainer):

    @dataclass
    class ScalingConfig:
        num_workers: int = 1
        num_cpus_per_worker: int = 1
        num_gpus_per_worker: int = 0
```

and add documentation to the expected values. I think this will make it clear for users what to pass for configuration here

Reply:
@krfricke we ended up deciding not to go with dataclasses for the ScalingConfig for now because it doesn't interoperate seamlessy with Tune. cc @ericl @richardliaw